src/zarr/codecs/_v2.py

```diff
@@ -83,6 +92,7 @@ async def _encode_single(
         else:
             cdata = chunk
 
+        cdata = ensure_contiguous(ensure_ndarray_like(cdata))
```

Comment:
an alternative is https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tobytes.html but I'm not sure if that always copies.

Reply:
`ensure_bytes` essentially calls `arr.tobytes` but adds a few checks.